modules/tensor_mechanics/src/materials/abaqus/AbaqusUmatMaterial.C

```
@@ -0,0 +1,262 @@
+  params.addRequiredParam<std::vector<Real>>("mechanical_constants",
+                                             "Mechanical Material Properties");
+  params.addParam<std::vector<Real>>("thermal_constants", "Thermal Material Properties");
+  params.addRequiredParam<unsigned int>("num_state_vars",
+                                        "The number of state variables this UMAT is going to use");
+  return params;
+}
+
+AbaqusUmatMaterial::AbaqusUmatMaterial(const InputParameters & parameters)
+  : ComputeStressBase(parameters),
+    _plugin(getParam<FileName>("plugin")),
+    _mechanical_constants(getParam<std::vector<Real>>("mechanical_constants")),
+    _thermal_constants(getParam<std::vector<Real>>("thermal_constants")),
+    _num_state_vars(getParam<unsigned int>("num_state_vars")),
+    _aqCELENT(1.0), // TODO: figure out what this really should be
+    _stress_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + "stress")),
+    _total_strain(getMaterialProperty<RankTwoTensor>(_base_name + "total_strain")),
+    _strain_increment(getMaterialProperty<RankTwoTensor>(_base_name + "strain_increment")),
+    _jacobian_mult(declareProperty<RankFourTensor>(_base_name + "Jacobian_mult")),
+    _Fbar(getMaterialPropertyOld<RankTwoTensor>(_base_name + "deformation_gradient")),
+    _Fbar_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + "deformation_gradient")),
+    _state_var(declareProperty<std::vector<Real>>(_base_name + "state_var")),
+    _state_var_old(getMaterialPropertyOld<std::vector<Real>>(_base_name + "state_var")),
+    _elastic_strain_energy(declareProperty<Real>(_base_name + "elastic_strain_energy")),
+    _plastic_dissipation(declareProperty<Real>(_base_name + "plastic_dissipation")),
+    _creep_dissipation(declareProperty<Real>(_base_name + "creep_dissipation")),
+    _material_timestep(declareProperty<Real>(_base_name + "material_timestep_limit"))
+{
+#if defined(METHOD)
+  _plugin += std::string("-") + QUOTE(METHOD) + ".plugin";
```

Just curious -- is METHOD always defined? Seems like this wouldn't even work if it weren't.

---

Yeah, we should probably just raise an error if it is not defined.